modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/RankFeaturesFieldMapper.java

```diff
@@ -41,12 +42,11 @@
     public static final String CONTENT_TYPE = "rank_features";
 
     public static class Defaults {
-        public static final MappedFieldType FIELD_TYPE = new RankFeaturesFieldType();
+        public static final FieldType FIELD_TYPE = new FieldType();
 
         static {
             FIELD_TYPE.setTokenized(false);
             FIELD_TYPE.setIndexOptions(IndexOptions.NONE);
-            FIELD_TYPE.setHasDocValues(false);
```

Comment:
This is because setting docvalues on lucene's FieldType doesn't have any effect at the API level, as adding a DocValues field explicitly will set things later.  But you do need to set index options separately.  A follow up will remove this entirely (RankFeaturesFieldMapper, along with many many other field mappers, doesn't actually need a lucene FieldType at all).

Reply:
thanks for clarifying